api/cluster/controller.go

```diff
@@ -291,3 +334,18 @@ func (c *controller) ListPods(ctx context.Context, namespace, labelSelector stri
 func (c *controller) StreamPodLogs(ctx context.Context, namespace, podName string, opts *corev1.PodLogOptions) (io.ReadCloser, error) {
 	return c.clusterClient.Pods(namespace).GetLogs(podName, opts).Stream(ctx)
 }
+
+func (c *controller) GetCurrentDeploymentScale(ctx context.Context, namespace, knRevisionName string) (int, error) {
```

Comment:
I think we can have this as private method since it's not being used by other package. Then we can remove it from Controller interface as well.

Reply:
I did consider that. The only downside to this is, it makes it harder to unit test it. Since the integration tests are also not able to test the initial scale annotation (as we don't return it from the [Deploy function](https://github.com/caraml-dev/merlin/blob/main/api/cluster/controller.go#L221)), I decided to keep it as a part of the interface and add a separate test case for it.